File: algorithms/ar-prfac/java/PrimeFactor.java
public class PrimeFactor{

    public static void primeFactors(int n){
        if(n<=1)
            return;
        for(int i=2;(i*i)<=n;i++){
            while(n%i==0){
                System.out.println(i);
                n=n/i;
            }
        }
        if(n>0){
            System.out.println(n);
        }
    }

    public static void main(String args[]){
        primeFactors(60);
    }
}

File: algorithms/ar-prfac/python3/prime_factor.py
def primefactor(arr):
    if arr<=0:
        return
    
    while arr%2==0:
        print(2)
        arr/=2

    for i in range(3, int(arr*arr)+1,2):

        while arr%i==0:
            print(i)
            arr/=i

    if arr>2:
        print(int(arr))


def main():
    primefactor(140)

    
if __name__ == '__main__':
    main()